src/models/room.ts

```diff
@@ -1862,7 +1862,7 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         shouldLiveInThread: boolean;
         threadId?: string;
     } {
-        if (!this.client.supportsExperimentalThreads()) {
+        if (!this.client?.supportsExperimentalThreads()) {
```

Comment:
Why was this change made, if the types say `client` is non-null? IMO we should either fix the places that pass in a falsy `client`, or change the type to match (though I strongly prefer the former)

Reply:
See https://github.com/matrix-org/matrix-js-sdk/pull/2848#discussion_r1013505393